src/DynamoCoreWpf/ViewModels/Watch3D/Watch3DViewModelBase.cs

```diff
@@ -0,0 +1,439 @@
+        }
+
+        protected virtual void OnRenderPackagesUpdated(NodeModel updatedNode, IEnumerable<IRenderPackage> packages)
+        {
+            OnBeginUpdate(packages);
+        }
+
+        public virtual void GenerateViewGeometryFromRenderPackagesAndRequestUpdate(
+            IEnumerable<IRenderPackage> taskPackages)
+        {
+            // Override in derived classes
+        }
+
+        protected virtual void OnNodePropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            // Override in derived classes.
+        }
+
+        internal virtual void ExportToSTL(string path, string modelName)
+        {
+            // Override in derived classes
+        }
+
+        /// <summary>
+        /// Display a label for one or several render packages 
+        /// based on the paths of those render packages.
+        /// </summary>
+        //public void TagRenderPackageForPath(string path)
+        //{
+        //    var packages = new List<IRenderPackage>();
+
+        //    //This also isn't thread safe
+        //    foreach (var node in model.CurrentWorkspace.Nodes)
+        //    {
+        //        lock (node.RenderPackagesMutex)
+        //        {
+        //            //Note(Luke): this seems really inefficent, it's doing an O(n) search for a tag
+        //            //This is also a target for memory optimisation
+
+        //            packages.AddRange(
+        //                node.RenderPackages.Where(x => x.Description == path || x.Description.Contains(path + ":")));
+        //        }
+        //    }
+
+        //    if (packages.Any())
+        //    {
+        //        //clear any labels that might have been drawn on this
+        //        //package already and add the one we want
+        //        if (currentTaggedPackages.Any())
+        //        {
+        //            currentTaggedPackages.ForEach(x => x.DisplayLabels = false);
+        //            currentTaggedPackages.Clear();
+        //        }
+
+        //        packages.ToList().ForEach(x => x.DisplayLabels = true);
+        //        currentTaggedPackages.AddRange(packages);
+
+        //        var allPackages = new List<IRenderPackage>();
+        //        var selPackages = new List<IRenderPackage>();
+
+        //        foreach (var node in model.CurrentWorkspace.Nodes)
+        //        {
+        //            lock (node.RenderPackagesMutex)
+        //            {
+        //                allPackages.AddRange(
+        //                    node.RenderPackages.Where(x => x.HasRenderingData && !x.IsSelected));
+        //                selPackages.AddRange(
+        //                    node.RenderPackages.Where(x => x.HasRenderingData && x.IsSelected));
+        //            }
+        //        }
+
+        //        //OnResultsReadyToVisualize(
+        //        //    new VisualizationEventArgs(allPackages, selPackages, Guid.Empty));
+        //    }
+        //}
```

Comment:
If we plan on keeping commented out code, let's add a task for why it's here. Otherwise it will be removed accidentally.